### ISSUE TITLE
Migrations: Parameterless DbContext constructor is used in preference to overload taking DbContextOptions

### ISSUE THREAD

**Issue by AndriySvyryd** [aspnet/EntityFramework#6826]
Using Startup.cs:

``` C#
public void ConfigureServices(IServiceCollection services)
{
    Console.WriteLine("ConfigureServices");

    services.AddDbContext<DataContext>(options =>
        options.UseSqlServer(Configuration.GetConnectionString(name: "DefaultConnection"))
            .EnableSensitiveDataLogging());
}
```

And context:

``` C#
public class DataContext : DbContext
{
    public DataContext()
    {
        Console.WriteLine("Parameterless ctor");
    }

    public DataContext(DbContextOptions<DataContext> options)
        : base(options)
    {
        Console.WriteLine("Options ctor");
    }
}
```

Output is

> ConfigureServices
> Parameterless ctor
> System.InvalidOperationException: No database provider has been configured for this DbContext. A provider can be configured by overriding the DbContext.OnConfiguring method or by using AddDbContext on the application service provider. If AddDbContext is used, then also ensure that your DbContext type accepts a DbContextOptions<TContext> object in its constructor and passes it to the base constructor for DbContext.

**Comment by bricelam** [comment]
It looks like we call `ActivatorUtilities.CreateInstance()`. I must have different semantics. We could try to call `IServiceProvider.GetService()` first and fallback...

**Comment by anpete** [comment]
@AndriySvyryd @bricelam @divega Is this a DI bug? Did anyone file it?

**Comment by divega** [comment]
Yes, someone did :smile: https://github.com/aspnet/DependencyInjection/issues/197, and it got moved to the backlog.

**Comment by divega** [comment]
> We could try to call IServiceProvider.GetService() first and fallback...

@bricelam FWIW, there is already `ActivatorUtilities.GetServiceOrCreateInstance()` that has those semantics.

**Comment by OskarKlintrot** [comment]
Should this be working in the latest (stable) release? I've the same behavior in `Microsoft.EntityFrameworkCore.Design` v1.1.2.

**Comment by bricelam** [comment]
@OskarKlintrot No, it was fixed in the nightly builds of 2.0.0 for a while, but got removed (by #8328) before 2.0.0-preview1.

Re-opening so we can consider putting the fix in 1.1.3.

**Comment by OskarKlintrot** [comment]
Thanks for the answer @bricelam, got a bit confused!

**Comment by streamtree** [comment]
Is there a workaround in the meantime? Previous comments are unclear on this.

**Comment by OskarKlintrot** [comment]
Don't know if there is a workaround @streamtree, I just removed the parameterless constructor.

**Comment by divega** [comment]
Removing labels since this fix doesn't belong in 2.0 anymore.

**Comment by ajcvickers** [comment]
@bricelam What is the status of this in 2.0? Is the DbContextOptions constructor used if present?

**Comment by bricelam** [comment]
It was re-fixed in 2.0 when we added support for `Program.BuildWebHost`.

**Comment by divega** [comment]
@bricelam ok, I misunderstood what you said at https://github.com/aspnet/EntityFramework/issues/6826#issuecomment-302133822. Then going ahead and closing this as fixed in preview2, and we can use #8825 to track porting it.
